examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+        schedule=TimeInterval(1.0),
+        mode = "c")
+
+# Build the `output_writer` for the growth rate, which is a scalar field.
+# Output every time step.
+#
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (perturbation_norm = perturbation_norm,),
+        filepath=joinpath(@__DIR__, "perturbation_norm_shallow_water.nc"),
+        schedule=IterationInterval(1),
+        dimensions=(perturbation_norm=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visualize the results
+#
+
+using NCDatasets
+nothing # hide
+
+# This sets the environment to avoid an error in GKSwstype
+
+ENV["GKSwstype"] = "100"
+
+# Define the coordinates for plotting
+
+xf = xnodes(ω_field)
+yf = ynodes(ω_field)
+nothing # hide
+
+# Define keyword arguments for plotting the contours
+
+kwargs = (
+         xlabel = "x",
+         ylabel = "y",
+           fill = true,
+         levels = 20,
+      linewidth = 0,
+          color = :balance,
+       colorbar = true,
+           ylim = (0, Ly),
+           xlim = (0, Lx)
+)
+nothing # hide
+
+# Read in the `output_writer` for the two-dimensional fields
+# and then create an animation showing both the total and perturbation
+# vorticities.
+
+using Plots
+
+ds = NCDataset(simulation.output_writers[:fields].filepath, "r")
+
+iterations = keys(ds["time"])
+
+anim = @animate for (iter, t) in enumerate(ds["time"])
+     ω = ds["ω"][:, :, 1, iter]
+    ωp = ds["ωp"][:, :, 1, iter]
+
+     ω_max = maximum(abs, ω)
+    ωp_max = maximum(abs, ωp)
+
+     plot_ω = contour(xf, yf, ω',  clim=(-ω_max,  ω_max),  title=@sprintf("Total ω at t = %.3f", t); kwargs...)
+    plot_ωp = contour(xf, yf, ωp', clim=(-ωp_max, ωp_max), title=@sprintf("Perturbation ω at t = %.3f", t); kwargs...)
+
+    plot(plot_ω, plot_ωp, layout = (1,2), size=(1200, 500))
+
+    print("At t = ", t, " maximum of ωp = ", maximum(abs, ωp), "\n")
+end
+
+close(ds)
+
+mp4(anim, "Bickley_Jet_ShallowWater.mp4", fps=15)
+
+# Read in the `output_writer` for the scalar field.
+
+ds2 = NCDataset(simulation.output_writers[:growth].filepath, "r")
+
+iterations = keys(ds2["time"])
+
+t = ds2["time"][:]
+σ = ds2["perturbation_norm"][:]
+
+close(ds2)
+
+# Import `Polynomials` to be able to use `best_fit`.
+# Compute the best fit slope and save the figure.
+
+using Polynomials
+
+I = 6000:7000
+best_fit = fit((t[I]), log.(σ[I]), 1)
+poly = 2 .* exp.(best_fit[0] .+ best_fit[1]*t[I])
+
+plt = plot(t[1000:end], log.(σ[1000:end]), lw=4, label="sigma", 
+        xlabel="time", ylabel="log(v)", title="growth of perturbation", legend=:bottomright)
+plot!(plt, t[I], log.(poly), lw=4, label="best fit")
+
+# We can compute the slope of the curve on a log scale, which approximates the growth rate
+# of the simulation. This should be close to the theoretical prediction.
+
+print("Growth rate in the simulation is approximated to be ", best_fit[1], 
+      ", which is close to the theoretical value of 0.14.")
```

Comment:
```suggestion
print("Growth rate in the simulation is approximated to be ", a, ",\n",
             "which is close to the theoretical value of 0.14.")
```

(splitting the output in two lines)

Reply:
Sure thing

Reply:
thanks